volmdlr/wires.py

```diff
@@ -621,10 +1079,14 @@ def __hash__(self):
         return sum([hash(p) for p in self.points])
 
     def __eq__(self, other_):
-        equal = True
-        for point, other_point in zip(self.points, other_.points):
-            equal = (equal and point == other_point)
-        return equal
+        if other_.__class__.__name__ != 'ClosedPolygon2D':
+           return False 
```

Comment:
better use other.__class__ is not self.__class__ and use it before point checking